crates/ruff_python_semantic/src/analyze/function_type.rs

```diff
@@ -11,6 +11,9 @@ pub enum FunctionType {
     Method,
     ClassMethod,
     StaticMethod,
+    // `__new__` is an implicit static method but
+    // is treated as a class method for some lint rules
```

Comment:
Nit: you could use doc-comments here so that we get nice tooltips when hovering over this method in VSCode 

```suggestion
    /// `__new__` is an implicit static method but
    /// is treated similarly to class methods for several lint rules
```